src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -177,10 +175,12 @@ public void Dispose()
 
         private (uint RefEscapeScope, uint ValEscapeScope) GetLocalScopes(LocalSymbol local)
         {
-            Debug.Assert(_localEscapeScopes?.ContainsKey(local) == true);
+            Debug.Assert(_localEscapeScopes?.ContainsKey(local) == true || _symbol != local.ContainingSymbol);
 
             return _localEscapeScopes?.TryGetValue(local, out var scopes) == true
                 ? scopes
+                : _symbol != local.ContainingSymbol // captured local
+                ? (CurrentMethodScope, CallingMethodScope)
                 : (CallingMethodScope, CallingMethodScope);
```

Comment:
>(CallingMethodScope, CallingMethodScope) [](http://example.com/codeflow?start=18&length=40)

Is this last value just a fallback in the unexpected case where the local is not in the dictionary and not captured? If so, then `CurrentMethodScope` seems like a safer value than `CallingMethodScope` for ref escape scope. In short, should we simply return `(CurrentMethodScope, CallingMethodScope)` whenever the local is not in the dictionary?